lambda-buffers-frontend/lambda-buffers-frontend.cabal

```diff
@@ -0,0 +1,135 @@
+cabal-version: 3.0
+name:          lambda-buffers-frontend
+version:       0.1.0.0
+synopsis:      Lambda Buffers Frontend
+
+-- license:
+
+author:        MLabs LTD
+maintainer:    info@mlabs.city
+
+-- A copyright notice.
+-- copyright:
+-- category:
+
+flag dev
+  description: Enable non-strict compilation for development
+  manual:      True
+
+common common-language
+  ghc-options:
+    -Wall -Wcompat -fprint-explicit-foralls -fprint-explicit-kinds
+    -fwarn-missing-import-lists -Weverything -Wno-unsafe
+    -Wno-missing-safe-haskell-mode -Wno-implicit-prelude
+    -Wno-missing-kind-signatures -Wno-all-missed-specializations
+
+  if !flag(dev)
+    ghc-options: -Werror
+
+  default-extensions:
+    NoStarIsType
+    BangPatterns
+    BinaryLiterals
+    ConstrainedClassMethods
+    ConstraintKinds
+    DataKinds
+    DeriveAnyClass
+    DeriveDataTypeable
+    DeriveFoldable
+    DeriveFunctor
+    DeriveGeneric
+    DeriveLift
+    DeriveTraversable
+    DerivingStrategies
+    DoAndIfThenElse
+    EmptyCase
+    EmptyDataDecls
+    EmptyDataDeriving
+    ExistentialQuantification
+    ExplicitForAll
+    ExplicitNamespaces
+    FlexibleContexts
+    FlexibleInstances
+    ForeignFunctionInterface
+    GADTSyntax
+    HexFloatLiterals
+    ImportQualifiedPost
+    InstanceSigs
+    KindSignatures
+    LambdaCase
+    MonomorphismRestriction
+    MultiParamTypeClasses
+    NamedFieldPuns
+    NamedWildCards
+    NumericUnderscores
+    OverloadedStrings
+    PartialTypeSignatures
+    PatternGuards
+    PolyKinds
+    PostfixOperators
+    RankNTypes
+    RelaxedPolyRec
+    ScopedTypeVariables
+    StandaloneDeriving
+    StandaloneKindSignatures
+    TemplateHaskell
+    TraditionalRecordSyntax
+    TupleSections
+    TypeApplications
+    TypeFamilies
+    TypeOperators
+    TypeSynonymInstances
+    ViewPatterns
+
+  default-language:   Haskell2010
+
+library
+  import:          common-language
+  build-depends:
+    , base           >=4.16
+    , containers     >=0.6
+    , directory      >=1.3
+    , filepath       >=1.4
+    , mtl            >=2.2
+    , parsec         >=3.1
+    , prettyprinter  >=1.7
+    , text           >=1.2
+    , transformers   >=0.5
+
+  hs-source-dirs:  src
+  exposed-modules:
+    LambdaBuffers.Frontend.FrontM
+    LambdaBuffers.Frontend.Parsec
+    LambdaBuffers.Frontend.PPrint
+    LambdaBuffers.Frontend.Syntax
+
+executable lambda-buffers-frontend-cli
+  import:         common-language
+  build-depends:
+    , base                        >=4.16
+    , containers                  >=0.6
+    , lambda-buffers-compiler-pb  >=0.1
+    , lambda-buffers-frontend
+    , lens                        >=5.2
+    , optparse-applicative        >=0.17
+    , prettyprinter               >=1.7
```

Comment:
```suggestion
  import:          common-deps
  build-depends:
    , lambda-buffers-compiler-pb  >=0.1
    , lambda-buffers-frontend
    , lens                        >=5.2
    , optparse-applicative        >=0.17
    , prettyprinter               >=1.7
```

Reply:
I'll stick with flat deps as I find myself finding deps intersections and performing tedious tasks. I think it's nice to refactor 'common' stuff into a stanza, but there's no tooling for that atm and it gets tedious imo. Additionally, automatic IDE package insertions don't respect this etc.